resolve/index.go

```diff
@@ -303,3 +347,80 @@ func passesLanguageFilter(langFilter []string, langName string) bool {
 	}
 	return false
 }
+
+const indexFileVersion = 1
+
+type indexFileData struct {
+	Version int           `json:"version"`
+	Data    []*ruleRecord `json:"data"`
+}
+
+// WriteToFile saves all data in the index to a JSON file at the path provided.
+//
+// This file is not expected to have a stable format between Gazelle versions, but
+// a best-effort attempt is made to keep it sorted so it _should_ remain stable
+// between runs.
```

Comment:
Let's comment for both `WriteToFile` and `ReadToFile` that these functions are experimental, and users outside Gazelle should not rely on them.

I'd also suggest not committing to JSON in the doc comment. This may not scale for very large repos: we might not want to read the file into memory. We might want to switch to SQLite or something similar. JSON is good for initial experimentation obviously though.